database-commons/src/main/java/io/cdap/plugin/db/batch/sink/CommonFieldsValidator.java

```diff
@@ -35,7 +35,7 @@
   protected static final Logger LOG = LoggerFactory.getLogger(CommonFieldsValidator.class);
 
   @Override
-  public void validateFields(Schema inputSchema, ResultSet resultSet) throws SQLException {
+  public Set<String> validateFields(Schema inputSchema, ResultSet resultSet) throws SQLException {
```

Comment:
should this take a FailureCollector instead of logging errors?

Reply:
This returns the set of invalid fields and the caller can decide to either log it or add to failure collector.

Reply:
but the caller doesn't know why the field is invalid, and this method already logs an error for each problem.

Reply:
done